addons/website_version/models/experiment.py

```diff
@@ -0,0 +1,300 @@
+    google_ref = fields.Char(string="Reference Google", required=True)
+
+
+class Experiment(models.Model):
+
+    _name = "website_version.experiment"
+    _inherit = ['mail.thread']
+    _order = 'sequence'
+
+    @api.multi
+    def _check_view(self):
+        #No overlap for running experiments
+        for exp in self:
+            if exp.state == 'running':
+                for exp_ver in exp.experiment_version_ids:
+                    if exp_ver.search([('version_id.view_ids.key', 'in', [v.key for v in exp_ver.version_id.view_ids]), ('experiment_id', '!=', exp_ver.experiment_id.id), ('experiment_id.website_id', '=', exp_ver.experiment_id.website_id.id), ('experiment_id.state', '=', 'running')]):
+                        return False
+        return True
+
+    @api.multi
+    def _check_website(self):
+        for exp in self:
+            for exp_ver in exp.experiment_version_ids:
+                if not exp_ver.version_id.website_id.id == exp.website_id.id:
+                    return False
+        return True
+
+    _constraints = [
+        (_check_view, _('This experiment contains a view which is already used in another running experience'), ['state']),
+        (_check_website, _('This experiment must have versions which are in the same website'), ['website_id', 'experiment_version_ids']),
+    ]
+
+    _group_by_full = {
+        'state': lambda *args, **kwargs: ([s[0] for s in [('running', 'Running'), ('paused', 'Paused'), ('ended', 'Ended')]], dict()),
+    }
+
+    def _get_version_number(self):
+        for exp in self:
+            exp.version_number = len(exp.experiment_version_ids) + 1
+
+    name = fields.Char(string="Title", required=True)
+    experiment_version_ids = fields.One2many('website_version.experiment_version', 'experiment_id', string="experiment_version_ids")
+    website_id = fields.Many2one('website', string="Website", required=True)
+    state = fields.Selection([('running', 'Running'), ('paused', 'Paused'), ('ended', 'Ended')], 'Status', required=True, copy=False, track_visibility='onchange', default='running')
+    goal_id = fields.Many2one('website_version.goals', string="Objective", required=True)
+    color = fields.Integer('Color Index')
+    version_number = fields.Integer(compute=_get_version_number)
+    sequence = fields.Integer('Sequence', required=True, help="Test.", default=1)
+    google_id = fields.Char(string="Google_id")
+
+    @api.model
+    def create(self, vals):
+        print vals
+        exp = {
+            'name': vals['name'],
+            'objectiveMetric': self.env['website_version.goals'].browse([vals['goal_id']])[0].google_ref,
+            'status': vals['state'],
+            'variations': [{'name': 'master', 'url': 'http://localhost/master'}]
+        }
+        version_list = vals.get('experiment_version_ids', [])
+        for version in version_list:
+            name = self.env['website_version.version'].browse([version[2]['version_id']])[0].name
```

Comment:
I already put that somewhere in the old PR / commit, but remember there are other commands than (6, 0, {}), for example (4, id).